worlds/kh2/Client.py

```diff
@@ -346,6 +346,29 @@ def on_package(self, cmd: str, args: dict):
         if cmd in {"DataPackage"}:
             if "Kingdom Hearts 2" in args["data"]["games"]:
                 self.data_package_kh2_cache(args)
+            if "KeybladeAbilities" in self.kh2slotdata.keys():
+                # sora ability to slot
+                self.AbilityQuantityDict.update(self.kh2slotdata["KeybladeAbilities"])
+                # itemid:[slots that are available for that item]
+                self.AbilityQuantityDict.update(self.kh2slotdata["StaffAbilities"])
+                self.AbilityQuantityDict.update(self.kh2slotdata["ShieldAbilities"])
+
+            all_weapon_location_id = []
+            for weapon_location in all_weapon_slot:
+                all_weapon_location_id.append(self.kh2_loc_name_to_id[weapon_location])
+            self.all_weapon_location_id = set(all_weapon_location_id)
```

Comment:
```suggestion
            self.all_weapon_location_id = {self.kh2_loc_name_to_id[loc] for loc in all_weapon_slot}
```
(abbreviated to fit 120 character limit)